src/m1_facilitator/handlers/DeclaredDepositIntentHandler.ts

```diff
@@ -0,0 +1,173 @@
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import BigNumber from 'bignumber.js';
+import DepositIntent from '../models/DepositIntent';
+import Message, { MessageStatus, MessageType } from '../models/Message';
+import GatewayRepository from '../repositories/GatewayRepository';
+import DepositIntentRepository from '../repositories/DepositIntentRepository';
+import MessageRepository from '../repositories/MessageRepository';
+import Utils from '../../common/Utils';
+import Logger from '../../m0_facilitator/Logger';
+
+/**
+ * It handles updates from DeclaredDepositIntents entity.
+ */
+export default class DeclaredDepositIntentsHandler {
+  /** Instance of DepositIntentRepository. */
+  private readonly depositIntentRepository: DepositIntentRepository;
+
+  /** Instance of MessageRepository. */
+  private readonly messageRepository: MessageRepository;
+
+  /** Instance of GatewayRepository. */
+  private readonly gatewayRepository: GatewayRepository;
+
+  /**
+   * Constructor for DeclaredDepositIntentHandler.
+   *
+   * @param depositIntentRepository Instance of DepositIntentRepository.
+   * @param gatewayRepository Instance of GatewayRepository.
+   * @param messageRepository Instance of MessageRepository.
+   */
+  public constructor(
+    depositIntentRepository: DepositIntentRepository,
+    gatewayRepository: GatewayRepository,
+    messageRepository: MessageRepository,
+  ) {
+    this.depositIntentRepository = depositIntentRepository;
+    this.gatewayRepository = gatewayRepository;
+    this.messageRepository = messageRepository;
+  }
+
+  /**
+   * It handles DeclaredDepositIntent entity records.
```

Comment:
Please add detailed documentation, how it handles. What this function does.

Reply:
Done